// Copyright 2021 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package nearby.sharing.proto;

option java_package = "com.google.android.gms.nearby.sharing";
option java_outer_classname = "CertificateMetadata";
option optimize_for = LITE_RUNTIME;

message EncryptedMetadata {
  // The name of the local device when certificate is created.
  optional string device_name = 1;

  // The name of the user whose device created the certificate.
  optional string full_name = 2;

  // The icon url of the user whose device created the certificate.
  optional string icon_url = 3;

  // The Bluetooth MAC address of the device which created the certificate.
  optional bytes bluetooth_mac_address = 4;

  // The obfuscated Gaia ID of the account which created the certificate.
  optional string obfuscated_gaia_id = 5;

  // The name of the account which created the certificate.
  optional string account_name = 6;

  // The device's model name
  optional string model_name = 7;

  // The vendor ID of the local device.
  optional int32 vendor_id = 8;
}
